src/java/picard/analysis/CollectAlignmentSummaryMetrics.java

```diff
@@ -116,6 +116,10 @@ public static void main(final String[] argv) {
                     "in the file are aligned then alignment summary metrics collection will fail.");
         }
 
+        if (super.REFERENCE_SEQUENCE != null && this.REFERENCE_SEQUENCE == null) {
```

Comment:
Unless I'm mistaken this.REFERENCE_SEQUENCE would only be null if someone was setting it to NULL explicitly since there is a default value.
 In that case we would be faithful to the fact that you want to override the Default as well as the super.REFERENCE_SEQUENCE to null. If that is not desired behavior we might want to consider what should be. In the mean time this will fix the issue.